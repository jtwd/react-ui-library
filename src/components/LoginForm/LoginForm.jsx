import React, { Component } from 'react'
import { node, func, bool } from 'prop-types'

import isEmailAddress from '../../services/validation/isEmailAddress'
import { validationMessages } from '../../config/systemMessages'
import Form from '../Form'
import TextInput from '../TextInput'
import PasswordInput from '../PasswordInput'
import Button from '../Button'
import FormControls from '../FormControls'

class LoginForm extends Component {
  constructor (props) {
    super(props)

    this.state = {
      fields: {
        email: '',
        password: ''
      },
      errors: {},
      submitted: false
    }
  }

  handleInputChange (evt) {
    const fields = this.state.fields;
    fields[evt.target.name] = evt.target.value;
    this.setState({fields});
  }

  handleFormSubmit (evt) {
    const {fields} = this.state
    const formIsValid = this.validate(fields)

    evt.preventDefault()

    if(formIsValid) {
      this.props.onSubmit(fields)
      this.setState({submitted: true})
    }
  }

  validate ({email, password}) {
    const errors = {}

    if (!email) {
      errors.email = validationMessages.email.required
    } else if (!isEmailAddress(email)) {
      errors.email = validationMessages.email.invalid
    }
    if (!password) errors.password = validationMessages.password.required

    this.setState({errors})
    const formIsValid = Object.getOwnPropertyNames(errors).length === 0
    return formIsValid
  }

  render () {
    const { errors } = this.state
    const { email, password } = this.state.fields
    const { title, focus } = this.props

    const Controls = (
      <FormControls>
        <Button primary icon="lock" type="submit">Login</Button>
      </FormControls>
    )

    return (
      <Form short collapse title={title} controls={Controls} onSubmit={(e) => this.handleFormSubmit(e)}>
        <TextInput
          htmlId="login-email"
          name="email"
          label="Email address"
          type="email"
          placeholder="user@domain.com"
          value={email}
          onChange={(e) => this.handleInputChange(e)}
          required
          error={errors.email}
          autoFocus={focus}
        />

        <PasswordInput
          htmlId="login-password"
          name="password"
          label="Password"
          value={password}
          onChange={(e) => this.handleInputChange(e)}
          maxLength={50}
          required
          error={errors.password} />

      </Form>
    )
  }
}

LoginForm.propTypes = {
  /** Form title - goes in the panel header */
  title: node,
  /** Auto focus on first input */
  focus: bool,
  /** Function to call when validation is passed */
  onSubmit: func.isRequired
}

LoginForm.defaultProps = {
  title: 'Login',
  focus: false
}

export default LoginForm
