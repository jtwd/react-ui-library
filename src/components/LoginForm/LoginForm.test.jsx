import React from 'react'
import {shallow} from 'enzyme'

import LoginForm from './LoginForm'

describe('LoginForm', () => {
  it('renders correctly', () => {
    const component = shallow(<LoginForm onSubmit={() => {}}/>)
    expect(component).toMatchSnapshot()
  })

})