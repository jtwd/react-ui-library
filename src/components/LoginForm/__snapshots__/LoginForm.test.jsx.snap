// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Wrapper
    app={false}
    centered={false}
    collapse={true}
    short={true}
    text={false}
>
    <Panel
        footer={
            <Button
                danger={false}
                icon="lock"
                large={false}
                onClick={[Function]}
                primary={false}
                secondary={true}
                small={true}
            >
                Login
            </Button>
        }
        footerRight={true}
        header="Login"
    >
        <TextInput
            error={null}
            htmlId="login-email"
            label="Email address"
            name="email"
            onChange={[Function]}
            placeholder="user@domain.com"
            required={true}
            type="email"
            value=""
        />
        <PasswordInput
            error=""
            htmlId="login-password"
            label="Password"
            maxLength={50}
            name="password"
            onChange={[Function]}
            placeholder=""
            required={true}
            showVisibilityToggle={true}
            value=""
        />
    </Panel>
</Wrapper>,
  "nodes": Array [
    <Wrapper
      app={false}
      centered={false}
      collapse={true}
      short={true}
      text={false}
>
      <Panel
            footer={
                  <Button
                        danger={false}
                        icon="lock"
                        large={false}
                        onClick={[Function]}
                        primary={false}
                        secondary={true}
                        small={true}
                  >
                        Login
                  </Button>
            }
            footerRight={true}
            header="Login"
      >
            <TextInput
                  error={null}
                  htmlId="login-email"
                  label="Email address"
                  name="email"
                  onChange={[Function]}
                  placeholder="user@domain.com"
                  required={true}
                  type="email"
                  value=""
            />
            <PasswordInput
                  error=""
                  htmlId="login-password"
                  label="Password"
                  maxLength={50}
                  name="password"
                  onChange={[Function]}
                  placeholder=""
                  required={true}
                  showVisibilityToggle={true}
                  value=""
            />
      </Panel>
</Wrapper>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LoginForm
        onSubmit={[Function]}
        title="Login"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LoginForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onSubmit": [Function],
          "title": "Login",
        },
        "refs": Object {},
        "state": Object {
          "credentials": Object {
            "email": "",
            "password": "",
          },
          "errors": Object {},
          "submitted": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Wrapper
          app={false}
          centered={false}
          collapse={true}
          short={true}
          text={false}
>
          <Panel
                    footer={
                              <Button
                                        danger={false}
                                        icon="lock"
                                        large={false}
                                        onClick={[Function]}
                                        primary={false}
                                        secondary={true}
                                        small={true}
                              >
                                        Login
                              </Button>
                    }
                    footerRight={true}
                    header="Login"
          >
                    <TextInput
                              error={null}
                              htmlId="login-email"
                              label="Email address"
                              name="email"
                              onChange={[Function]}
                              placeholder="user@domain.com"
                              required={true}
                              type="email"
                              value=""
                    />
                    <PasswordInput
                              error=""
                              htmlId="login-password"
                              label="Password"
                              maxLength={50}
                              name="password"
                              onChange={[Function]}
                              placeholder=""
                              required={true}
                              showVisibilityToggle={true}
                              value=""
                    />
          </Panel>
</Wrapper>,
        "_debugID": 2,
        "_renderedOutput": <Wrapper
          app={false}
          centered={false}
          collapse={true}
          short={true}
          text={false}
>
          <Panel
                    footer={
                              <Button
                                        danger={false}
                                        icon="lock"
                                        large={false}
                                        onClick={[Function]}
                                        primary={false}
                                        secondary={true}
                                        small={true}
                              >
                                        Login
                              </Button>
                    }
                    footerRight={true}
                    header="Login"
          >
                    <TextInput
                              error={null}
                              htmlId="login-email"
                              label="Email address"
                              name="email"
                              onChange={[Function]}
                              placeholder="user@domain.com"
                              required={true}
                              type="email"
                              value=""
                    />
                    <PasswordInput
                              error=""
                              htmlId="login-password"
                              label="Password"
                              maxLength={50}
                              name="password"
                              onChange={[Function]}
                              placeholder=""
                              required={true}
                              showVisibilityToggle={true}
                              value=""
                    />
          </Panel>
</Wrapper>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginForm
    onSubmit={[Function]}
    title="Login"
/>,
}
`;
